models/pull.go

```diff
@@ -97,6 +98,25 @@ func init() {
 	db.RegisterModel(new(PullRequest))
 }
 
+func deletePullsByBaseRepoID(sess db.Engine, repoID int64) error {
+	deleteCond := builder.Select("id").From("pull_request").Where(builder.Eq{"pull_request.base_repo_id": repoID})
+
+	// Delete scheduled auto merges
+	if _, err := sess.In("pull_id", deleteCond).
+		Delete(&pull_model.AutoMerge{}); err != nil {
+		return err
+	}
+
+	// Delete review states
+	if _, err := sess.In("pull_id", deleteCond).
+		Delete(&pull_model.ReviewState{}); err != nil {
+		return err
+	}
```

Comment:
IIRC there is no index for column ReviewState.PullID. It might affect performance 

well, deleting repo is a rare case, so the slow SQL here might be acceptable.

Reply:
@wxiaoguang it's a combined index with the pull_id as the second key.

Reply:
NO, the second key will never be used for this query in standard. No matter what database you use.

You can try to write an database engine to know why it couldn't.

Reply:
Ah okay, yes then they are indeed not indexed.

Reply:
Should I add it?

Reply:
> well, deleting repo is a rare case, so the slow SQL here might be acceptable.

either is fine to me, maybe we can leave a comment there (code for deleting) in case someone in future need such information.